File: backend/src/main/java/com/penocean/ehs/model/AccessRequest.java
package com.penocean.ehs.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AccessRequest {

    private Long id;
    private String requestNo;           // AR-YYYYMMDD-NNNN
    private Long companyId;
    private Long vesselId;
    private Long portId;
    private String workType;
    private String workDescription;
    private LocalDate plannedStartDate;
    private LocalDate plannedEndDate;
    private Integer workerCount;
    private String status;              // DRAFT/SUBMITTED/IN_REVIEW/IMPROVEMENT_REQUESTED/APPROVED/REJECTED
    private Long submittedBy;
    private LocalDateTime submittedAt;
    private Long reviewedBy;
    private LocalDateTime reviewedAt;
    private String improvementRequestReason;
    // [2026-04-23] PPT 슬라이드 14: 안전담당자 정보
    private String safetyManagerName;
    private String safetyManagerTel;
    private String safetyManagerEmail;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Boolean deleted;
}

File: frontend/src/types/accessRequest.ts
// Backend 계약: /api/access-requests
import type { PageResponse } from './common'

export type AccessRequestStatus =
  | 'DRAFT'
  | 'SUBMITTED'
  | 'IN_REVIEW'
  | 'IMPROVEMENT_REQUESTED'
  | 'APPROVED'
  | 'REJECTED'

export type AttachmentType =
  | 'RISK_ASSESSMENT'
  | 'PLEDGE'
  | 'WORK_PLAN'
  | 'OTHER'

export type ReviewAction =
  | 'START'
  | 'IMPROVEMENT'
  | 'APPROVE'
  | 'REJECT'

export interface AccessRequestListItem {
  id: number
  requestNo: string
  companyName: string
  vesselName: string
  portName: string | null      // [2026-04-23] PPT 슬라이드 13: 지역/항구
  industryName: string | null  // [2026-04-23] PPT 슬라이드 13: 업종
  workType: string
  plannedStartDate: string
  plannedEndDate: string
  workerCount: number
  status: AccessRequestStatus
  submittedAt: string | null
  improvementRequestReason?: string | null
}

export interface AccessRequestWorker {
  id?: number
  workerName: string
  workerBirth?: string
  workerPhone?: string
  workerRole?: string
  safetyEduCompleted: boolean
  safetyEduCompletedAt?: string
  safetyEduCertificateUrl?: string
  checkByShip?: boolean
  checkByShipAt?: string | null
}

export interface AccessRequestAttachment {
  id: number
  attachmentType: AttachmentType
  fileName: string
  filePath: string
  fileSize: number
  mimeType: string
  uploadedAt: string
}

export interface AccessRequestReviewLog {
  id: number
  action: ReviewAction
  comment?: string
  actorUserName?: string
  actedAt: string
}

export interface AccessRequestDetailResponse extends AccessRequestListItem {
  companyId: number
  vesselId: number
  portId?: number
  workDescription: string
  // [2026-04-23] PPT 슬라이드 14: 안전담당자 정보
  safetyManagerName?: string | null
  safetyManagerTel?: string | null
  safetyManagerEmail?: string | null
  noRiskAssessment?: boolean
  workers: AccessRequestWorker[]
  attachments: AccessRequestAttachment[]
  reviewLogs: AccessRequestReviewLog[]
}

export interface AccessRequestListParams {
  status?: AccessRequestStatus | ''
  companyId?: number
  vesselId?: number
  keyword?: string
  dateFrom?: string
  dateTo?: string
  page?: number
  size?: number
}

export interface CreateAccessRequestPayload {
  vesselId: number
  portId?: number
  companyId?: number
  workType: string
  workDescription: string
  plannedStartDate: string
  plannedEndDate: string
  // [2026-04-23] PPT 슬라이드 14: 안전담당자 정보
  safetyManagerName?: string | null
  safetyManagerTel?: string | null
  safetyManagerEmail?: string | null
  noRiskAssessment?: boolean
  workers?: Array<Omit<AccessRequestWorker, 'id' | 'safetyEduCertificateUrl'>>
}

export interface UpdateAccessRequestPayload extends CreateAccessRequestPayload {}

export interface ReviewPayload {
  action: ReviewAction
  comment?: string
}

export interface AddWorkersPayload {
  workers: Array<Omit<AccessRequestWorker, 'id' | 'safetyEduCertificateUrl'>>
}

export type AccessRequestPage = PageResponse<AccessRequestListItem>
